File: app/src/main/java/com/recursia/popularmovies/di/AppComponent.java
package com.recursia.popularmovies.di;

import com.recursia.popularmovies.di.modules.InteractorModule;
import com.recursia.popularmovies.domain.DetailScreenInteractor;
import com.recursia.popularmovies.domain.FavoriteScreenInteractor;
import com.recursia.popularmovies.domain.MoviesListInteractor;
import com.recursia.popularmovies.domain.SearchScreenInteractor;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {InteractorModule.class})
public interface AppComponent {
    DetailScreenInteractor getDetailScreenInteractor();

    FavoriteScreenInteractor getFavoriteScreenInteractor();

    MoviesListInteractor getMoviesListInteractor();

    SearchScreenInteractor getSearchScreenInteractor();
}

File: app/src/main/java/com/recursia/popularmovies/di/AppComponent.kt
package com.recursia.popularmovies.di

import com.recursia.popularmovies.di.modules.InteractorModule
import com.recursia.popularmovies.di.modules.NavigationModule
import com.recursia.popularmovies.domain.*
import dagger.Component
import ru.terrakok.cicerone.NavigatorHolder
import ru.terrakok.cicerone.Router
import javax.inject.Singleton

@Singleton
@Component(modules = [InteractorModule::class, NavigationModule::class])
interface AppComponent {
    val detailScreenInteractor: DetailScreenInteractor

    val mainScreenInteractor: MainScreenInteractor

    val searchScreenInteractor: SearchScreenInteractor

    val accountScreenInteractor: AccountScreenInteractor

    val genresBottomSheetInteractor: GenresBottomSheetInteractor

    val navigationHolder: NavigatorHolder

    val router: Router
}
